Sources/Apollo/InMemoryNormalizedCache.swift

```diff
@@ -2,6 +2,7 @@ import Foundation
 
 public final class InMemoryNormalizedCache: NormalizedCache {
   private var records: RecordSet
+  private let recordsLock = NSLock()
```

Comment:
Is there a reason why we aren't using `NSRecursiveLock` here? If the implementation of accessing the records ever changes to need to access the cache recursively or anything, then this is going to deadlock. 

If there is a good reason I'm missing to use a regular `NSLock`, then great, but I prefer to default to recursive if possible for safety.

Reply:
no good reason, switched